artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/converter/AMQPMessageIdHelper.java

```diff
@@ -130,6 +130,8 @@ public String toCorrelationIdString(Object idObject) {
             // It has "ID:" prefix and doesn't have encoding prefix, use it as-is.
             return stringId;
          }
+      } else if (idObject instanceof Binary) {
+         return ((Binary)idObject).getArray();
```

Comment:
Strictly speaking, its possible the array isnt just the id...the Binary should be checked that it doesnt have an array offset and is the same length as the array.

Reply:
I think I get what you mean. Check the latest push.